DOCS-10222 RTF 1.8.x

hegderohit89 opened the pull request.

hegderohit89 committed:
GKE starter doc for RTF

hegderohit89 commented on modules/ROOT/pages/index-self-managed.adoc:
```diff
@@ -1,14 +1,14 @@
 == Architecture
 
-Runtime Fabric on Self-Managed Kubernetes and its related components run as services on your EKS or AKS environment. The following diagram shows different layers of the architecture.
+Runtime Fabric on Self-Managed Kubernetes and its related components run as services on your EKS or AKS or GKE environment. The following diagram shows different layers of the architecture.
```
We have a lot of statements with `AKS or EKS or GKE`
Do we want to update it as `AKS/EKS/GKE`

sleslie23 replied:
No, in this case it would have to be "... EKS, AKS, or GKE environments."

kmendiratta-crm committed:
Update docs

kmendiratta-crm committed:
Update self managed documentation

kmendiratta-crm committed:
Update self managed documentation

Commit:
DOCS-10313 minor edits.

Commit:
DOCS-10313 adding GKS to other topics.

sundippatel committed:
Update index-self-managed.adoc

Engines -> Engine

sarjeet2013 commented on modules/ROOT/pages/index-self-managed.adoc:
```diff
@@ -45,15 +45,15 @@ Runtime Fabric on Self-Managed Kubernetes requires worker nodes which use the x8
 === Ingress Controller
 
-Runtime Fabric on Self-Managed Kubernetes supports any Ingress controller that is compatible with the Kubernetes environment on your vendor (EKS or AKS).
+Runtime Fabric on Self-Managed Kubernetes supports any Ingress controller that is compatible with the Kubernetes environment on your cloud infrastructure vendor.
```
@sundippatel Do we want to explicitly call out GCP based ingress controller not supported for GKE? Or, modify the current statement to let customer tweaks configuration for specific ingress controller to fit their needs.

sundippatel replied:
Agree, we should modify to clarify ingress controllers exhibiting the behavior that GKE's default ingress controller is are not supported.

sundippatel replied:
Attempted to address; please review.
If able, we'd like to point to a KB article that we'll need to publish. Added a placeholder for now.

sarjeet2013 commented on modules/ROOT/pages/index.adoc:
```diff
@@ -18,7 +18,7 @@ Contact your Customer Success Manager for information on how enable Anypoint Run
 There are two management options for Anypoint Runtime Fabric:
 
-* *Runtime Fabric on Self-Managed Kubernetes*: a version of Runtime Fabric that you install on an existing Kubernetes environment that you operate and manage. This version supports Amazon Elastic Kubernetes Service (EKS) and Azure Kubernetes Service (AKS).
+* *Runtime Fabric on Self-Managed Kubernetes*: a version of Runtime Fabric that you install on an existing Kubernetes environment that you operate and manage. This version supports Amazon Elastic Kubernetes Service (EKS), Azure Kubernetes Service (AKS), and Google Kubernetes Engines (GKE).
```
nit: Google Kubernetes Engines -> Google Kubernetes Engine

sleslie23 replied:
Fixed

sarjeet2013 commented on modules/ROOT/pages/install-azure.adoc:
```diff
@@ -4,7 +4,7 @@ include::_attributes.adoc[]
 :noindex:
 
-This topic describes how to install Anypoint Runtime Fabric on VMs / Bare Metal on your Azure account. Note that you can also use AKS to install xref:index-self-managed.adoc[Runtime Fabric on Self-Managed Kubernetes].
+This topic describes how to install Anypoint Runtime Fabric on VMs / Bare Metal on your Azure account. Note that you can also use Elastic Kubernetes Service (EKS), Azure Kubernetes Service (AKS), or Google Kubernetes Engines (GKE) to install xref:index-self-managed.adoc[Runtime Fabric on Self-Managed Kubernetes].
```
nit: Google Kubernetes Engines -> Google Kubernetes Engine

sleslie23 replied:
Fixed!

sarjeet2013 commented on modules/ROOT/pages/install-self-managed.adoc:
```diff
@@ -5,7 +5,7 @@
 ====
 
-Anypoint Runtime Fabric can be installed and operated on a AWS Elastic Kubernetes Service (EKS) or Azure Kubernetes Service (AKS) that you manage.
+Anypoint Runtime Fabric can be installed and operated on a AWS Elastic Kubernetes Service (EKS), Azure Kubernetes Service (AKS) or Elastic Kubernetes Service (EKS), Azure Kubernetes Service (AKS), or Google Kubernetes Engines (GKE) that you manage.
```
nit: Google Kubernetes Engines -> Google Kubernetes Engine

sleslie23 replied:
Fixed.

sundippatel committed:
Update index.adoc

Engines -> Engine

sundippatel committed:
Update index-self-managed.adoc

Added clarification on which ingress controllers are supported, with callout to GKE.
TODO: Add KB link.

sundippatel committed:
Update install-azure.adoc

Engines -> Engine

sundippatel commented:
![Screen Shot 2021-03-11 at 12 24 54 PM](https://user-images.githubusercontent.com/2469588/110835831-27b57800-8254-11eb-9341-046c9774e578.png)

@sleslie23 On the index-self-managed page, could we update the architecture diagram with this one, which adds in GKE?

sundippatel committed:
Update custom-ingress-configuration.adoc

Added note for GKE customers.

Commit:
Added Google Cloud when Azure or AWS are mentioned.

Commit:
Shared responsibility for Self-Managed Kubernetes

Commit:
minor adjustment to simplify ingress controller support.

Commit:
Added section - how ingress resources are generated; other minor edits

Commit:
minor edits

Commit:
added KB article for GKE ingress behavior

Commit:
Missing GKE on index page

Commit:
formatting issues with unordered lists

Commit:
Merge branch 'v1.8' into feat/docs-10262

sundippatel committed:
Update index-self-managed.adoc

Clarify responsibility of default and custom ingress resources

Commit:
DOCS-10222 copy edit.

Commit:
DOCS-10222 editing for consistency with style guide.

Commit:
DOCS-10261 adding info about deprecated port.

Commit:
DOCS-10261 adding info about deprecated port.

Commit:
DOCS-10261 clarifying content around deprecated port.

Commit:
DOCS-10440 updating graphics for GKE.

sundippatel commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -194,8 +194,14 @@ Because the following endpoints use mutual TLS authentication to establish the c
 .2+| 443 .2+| AMQP over WebSockets | US control plane: `transport-layer.prod.cloudhub.io` | Runtime Fabric message broker for interaction with the control plane.
 | EU control plane: `transport-layer.prod-eu.msap.io` | Runtime Fabric message broker for interaction with the control plane.
-.2+| 5044 .2+| TCP (Lumberjack) | US control plane: `dias-ingestor-nginx.prod.cloudhub.io` | Anypoint Monitoring agent for Runtime Fabric.
+.2+| 5044 (deprecated) .2+| TCP (Lumberjack) | US control plane: `dias-ingestor-nginx.prod.cloudhub.io` | Anypoint Monitoring agent for Runtime Fabric.
+
+*This port and hostname are deprecated in Anypoint Runtime Fabric, version 1.8.48.* 
+
+If you are using a previous version of Anypoint Runtime Fabric you must add this port and hostname to your allow list. If you are using a newer version, use the port and hostname specified below.
```
@sleslie23 Is there a way to place this note so that it applies for both the US and EU control plane endpoints?
Otherwise LGTM

sleslie23 replied:
I can add something like "This is applicable to endpoints in both the US and EU clouds." It's better to avoid trying change the format of the table rows.

sundippatel commented on modules/ROOT/pages/custom-ingress-configuration.adoc:
```diff
@@ -2,14 +2,29 @@
+
+By default, each application deployment has its own service and ingress resources. 
+
+In Runtime Fabric v1.8.50 and later, annotations and labels can be added to the generated Service resources. These changes persist for the lifecycle of those resources.
```
@sarjeet2013 Could we update this with the right agent version? (I just put down v1.8.50 as a placeholder).

sarjeet2013 replied:
v1.8.50 is correct.

sundippatel reviewed (dismissed):
Left a couple of comments inline; otherwise LGTM

sarjeet2013 commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -194,8 +194,14 @@ Because the following endpoints use mutual TLS authentication to establish the c
-.2+| 5044 .2+| TCP (Lumberjack) | US control plane: `dias-ingestor-nginx.prod.cloudhub.io` | Anypoint Monitoring agent for Runtime Fabric.
+.2+| 5044 (deprecated) .2+| TCP (Lumberjack) | US control plane: `dias-ingestor-nginx.prod.cloudhub.io` | Anypoint Monitoring agent for Runtime Fabric.
+
+*This port and hostname are deprecated in Anypoint Runtime Fabric, version 1.8.48.* 
```
nit: `1.8.50+` or `1.8.50 or later`

sleslie23 replied:
Fixed.

sarjeet2013 commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -194,8 +194,14 @@ Because the following endpoints use mutual TLS authentication to establish the c
+
+If you are using a previous version of Anypoint Runtime Fabric you must add this port and hostname to your allow list. If you are using a newer version, use the port and hostname specified below.
 | EU control plane: `dias-ingestor-nginx.prod-eu.msap.io` | Anypoint Monitoring agent for Runtime Fabric.
+.2+| 443 (v1.8.48, or later) .2+| TCP (Lumberjack) | US control plane: `dias-ingestor-router.us-east-1.prod.cloudhub.io` | Anypoint Monitoring agent for Runtime Fabric.
```
nit: v1.8.50, or later

sleslie23 replied:
Fixed.

Commit:
DOCS-10222 incorporate final review comments.

Mulesoft-Mysti commented on modules/ROOT/pages/custom-ingress-configuration.adoc:
```diff
@@ -2,14 +2,29 @@
+The template file is a Kubernetes ConfigMap file. After customizing the template, use the `rtfctl apply ingress-configmap` command to apply the custom Ingress configuration to your Kubernetes cluster.
 
-If you do not apply a custom ConfigMap file based on a template file, Runtime Fabric will default to a standard ingress resource definition. Depending on your ingress controller, this resource definition may not work. In this case, you must apply a custom ConfigMap using the template.
+If you do not apply a custom ConfigMap file based on a template file, Runtime Fabric defaults to a standard ingress resource definition. Depending on your ingress controller, this resource definition may not work. In this case, you must apply a custom ConfigMap using the template.
```
```suggestion
If you do not apply a custom ConfigMap file based on a template file, Runtime Fabric defaults to a standard Ingress resource definition. Depending on your ingress controller, this resource definition may not work. In this case, you must apply a custom ConfigMap using the template.
```

Mulesoft-Mysti replied:
ingress is capped in line 8.

Mulesoft-Mysti commented on modules/ROOT/pages/custom-ingress-configuration.adoc:
```diff
@@ -2,14 +2,29 @@
+
+== How Ingress Resources are Generated
+
+When deploying a Mule application to Runtime Fabric on Self-Managed Kubernetes, the Runtime Fabric agent generates a service and ingress resource for that application. The Runtime Fabric agent creates these Kubernetes resources in a namespace that corresponds to the Anypoint Environment ID used when deploying the Mule application.
```
```suggestion
When deploying a Mule application to Runtime Fabric on Self-Managed Kubernetes, the Runtime Fabric agent generates a service and Ingress resource for that application. The Runtime Fabric agent creates these Kubernetes resources in a namespace that corresponds to the Anypoint Environment ID used when deploying the Mule application.
```

Mulesoft-Mysti commented on modules/ROOT/pages/custom-ingress-configuration.adoc:
```diff
@@ -2,14 +2,29 @@
+
+When deploying a Mule application to Runtime Fabric on Self-Managed Kubernetes, the Runtime Fabric agent generates a service and ingress resource for that application. The Runtime Fabric agent creates these Kubernetes resources in a namespace that corresponds to the Anypoint Environment ID used when deploying the Mule application.
+
+By default, each application deployment has its own service and ingress resources. 
```
```suggestion
By default, each application deployment has its own service and Ingress resources. 
```

Mulesoft-Mysti commented on modules/ROOT/pages/custom-ingress-configuration.adoc:
```diff
@@ -2,14 +2,29 @@
+
+In Runtime Fabric v1.8.50 and later, annotations and labels can be added to the generated Service resources. These changes persist for the lifecycle of those resources.
+
+The generated ingress resources can be modified by using the ingress template. If further customization is required, follow best practices for disabling Ingress resource generation and managing the Ingress resources separately. To disable ingress resource generation, uncheck the *Enable inbound requests* option under the *Ingress* tab when deploying a Mule application in Anypoint Runtime Manager.
```
```suggestion
The generated ingress resources can be modified by using the Ingress template. If further customization is required, follow best practices for disabling Ingress resource generation and managing the Ingress resources separately. To disable Ingress resource generation, uncheck the *Enable inbound requests* option under the *Ingress* tab when deploying a Mule application in Anypoint Runtime Manager.
```

Mulesoft-Mysti commented on modules/ROOT/pages/custom-ingress-configuration.adoc:
```diff
@@ -2,14 +2,29 @@
+
+[IMPORTANT]
+====
+On Google Kubernetes Engine (GKE), the ingress controller included with GKE provisions a separate HTTP load balancer per application by default. Before using the ingress controller provided by GKE, learn more about its behavior, exploring workarounds, or use another ingress controller if this behavior is undesirable. See link:https://help.mulesoft.com/s/article/Default-Ingress-Controller-Behavior ...
```
```suggestion
On Google Kubernetes Engine (GKE), the Ingress controller included with GKE provisions a separate HTTP load balancer per application by default. Before using the Ingress controller provided by GKE, learn more about its behavior, exploring workarounds, or use another Ingress controller if this behavior is undesirable. See link:https://help.mulesoft.com/s/article/Default-Ingress-Controller-Behavior-with-Runtime-Fabric-on-GKE[KB article] for more details.
```

Mulesoft-Mysti commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -39,27 +39,27 @@ Knowledge in the following areas is required to install Runtime Fabric on VMs /
-* Creating separate Runtime Fabrics for production and non-production applications
-* Creating separate Runtime Fabrics based upon environment network scope, such as DMZ versus private network 
-* Ensuring that the Runtime Fabric default pod and services CIDR block do not interfere with the regions network
+* Creating separate runtime rabrics for production and non-production applications
```
```suggestion
* Creating separate runtime fabrics for production and non-production applications
```

Mulesoft-Mysti commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -39,27 +39,27 @@ Knowledge in the following areas is required to install Runtime Fabric on VMs /
-* Creating separate Runtime Fabrics based upon environment network scope, such as DMZ versus private network 
-* Ensuring that the Runtime Fabric default pod and services CIDR block do not interfere with the regions network
+* Creating separate runtime rabrics for production and non-production applications
+* Creating separate runtime rabrics based upon environment network scope, such as DMZ versus private network 
```
```suggestion
* Creating separate runtime fabrics based upon environment network scope, such as DMZ versus private network 
```

Mulesoft-Mysti commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -39,27 +39,27 @@ Knowledge in the following areas is required to install Runtime Fabric on VMs /
-* Ensuring that the Runtime Fabric default pod and services CIDR block do not interfere with the regions network
+* Creating separate runtime rabrics for production and non-production applications
+* Creating separate runtime rabrics based upon environment network scope, such as DMZ versus private network 
+* Ensuring that the runtime fabric default pod and services CIDR block do not interfere with the regions network
```
should be "regional network" or "region's network" (I think!)

Commit:
Update modules/ROOT/pages/install-prereqs.adoc

Co-authored-by: Mysti at Mulesoft <39537186+Mulesoft-Mysti@users.noreply.github.com>

Commit:
Update modules/ROOT/pages/install-prereqs.adoc

Co-authored-by: Mysti at Mulesoft <39537186+Mulesoft-Mysti@users.noreply.github.com>
